pkgs/development/libraries/physics/yoda/default.nix

```diff
@@ -9,14 +9,26 @@ stdenv.mkDerivation rec {
     sha256 = "1ki88rscnym0vjxpfgql8m1lrc7vm1jb9w4jhw9lvv3rk84lpdng";
   };
 
-  pythonPath = []; # python wrapper support
-
-  buildInputs = with python2Packages; [ python numpy matplotlib makeWrapper ]
+  patches = [
+    # fixes "TypeError: expected bytes, str found" in writeYODA()
+    (fetchpatch {
+      url = "https://gitlab.com/hepcedar/yoda/commit/d2bbbe92912457f8a29b440cbfa0b39daf28ec34.diff";
+      sha256 = "1x60piswpxwak61r2sdclsc8pzi1fshpkjnxlyflsa1iap77vkq8";
+    })
+  ];
+
+  nativeBuildInputs = with python.pkgs; [ cython makeWrapper ];
+  buildInputs = [ python ]
+    ++ (with python.pkgs; [ numpy matplotlib ])
     ++ stdenv.lib.optional withRootSupport root;
   propagatedBuildInputs = [ zlib ];
 
   enableParallelBuilding = true;
 
+  postPatch = ''
+    touch pyext/yoda/*.{pyx,pxd}
+  '';
```

Comment:
can you comment why this is necessary?

I'm assuming this is to fix some issue with cython?

Reply:
This is needed to regenerate the cython C++ code  (remember, this is built with "make"), the one in the release tarball doesn't support python3.7.

Reply:
ah, okay, force it to recompile. this makes sense